frontend/Interfaces/buddy/DIP/imgcodecs/loadsave.hpp

```diff
@@ -0,0 +1,220 @@
+static ImageCodecInitializer<T, N> &getCodecs() {
+#ifdef CV_CXX11
+  static ImageCodecInitializer<T, N> g_codecs;
+  return g_codecs;
+#else
+  // C++98 doesn't guarantee correctness of multi-threaded initialization of
+  // static global variables (memory leak here is not critical, use C++11 to
+  // avoid that)
+  static ImageCodecInitializer<T, N> *g_codecs =
+      new ImageCodecInitializer<T, N>();
+  return *g_codecs;
+#endif
+}
+
+/**
+ * Find the decoders
+ *
+ * @param[in] filename File to search
+ *
+ * @return Image decoder to parse image file.
+ */
+template <typename T, size_t N>
+static std::unique_ptr<BaseImageDecoder<T, N>>
+findDecoder(const String &filename) {
+
+  size_t i, maxlen = 0;
+
+  /// iterate through list of registered codecs
+  ImageCodecInitializer<T, N> &codecs = getCodecs<T, N>();
+  for (i = 0; i < codecs.decoders.size(); i++) {
+    size_t len = codecs.decoders[i]->signatureLength();
+    maxlen = std::max(maxlen, len);
+  }
+
+  /// Open the file
+  FILE *f = fopen(filename.c_str(), "rb");
+
+  /// in the event of a failure, return an empty image decoder
+  if (!f) {
+    return nullptr;
+  }
+  // read the file signature
+  String signature(maxlen, ' ');
+  maxlen = fread((void *)signature.c_str(), 1, maxlen, f);
+  fclose(f);
+  signature = signature.substr(0, maxlen);
+
+  /// compare signature against all decoders
+  for (i = 0; i < codecs.decoders.size(); i++) {
+    if (codecs.decoders[i]->checkSignature(signature))
+
+      return codecs.decoders[i]->newDecoder();
+  }
+  /// If no decoder was found, return base type
+  return nullptr;
+}
+
+template <typename T, size_t N>
+Img<T, N> imread(const String &filename, int flags) {
+  std::unique_ptr<BaseImageDecoder<T, N>> decoder = findDecoder<T, N>(filename);
+  if (decoder) {
+    // Converts a pointer to BmpDecoder<T, N>
+    BmpDecoder<T, N> *bmpDecoderPtr =
+        dynamic_cast<BmpDecoder<T, N> *>(decoder.get());
+    if (bmpDecoderPtr) {
+      // After creating the BmpDecoder<T, N> instance, perform related
+      // operations Defines whether the image is scaled or not
+      int scale_denom = 1;
+      bmpDecoderPtr->setScale(scale_denom);
+      // Set image path
+      bmpDecoderPtr->setSource(filename);
+      // Read image head
+      bmpDecoderPtr->readHeader();
+      _Size size(bmpDecoderPtr->width(), bmpDecoderPtr->height());
+      // grab the decoded type
+      int type = bmpDecoderPtr->type();
+      if ((flags & IMGRD_ANYDEPTH) == 0) {
+        type = IMG_MAKETYPE(IMG_8U, IMG_MAT_CN(type));
+      }
+      if ((flags & IMGRD_COLOR) != 0 ||
+          ((flags & IMGRD_ANYCOLOR) != 0 && IMG_MAT_CN(type) > 1)) {
+        type = IMG_MAKETYPE(IMG_MAT_DEPTH(type), 3);
+      } else {
+        type = IMG_MAKETYPE(IMG_MAT_DEPTH(type), 1);
+      }
+      // Create an Img instance
+      Img<T, N> Image;
+      Image.create(size.height, size.width, type);
+      bmpDecoderPtr->readData(Image);
+      return Image;
+    }
+  }
```

Comment:
What if decoder doesn't exist? Control reaches end of non-void function.
Consider throw an error here.